src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -731,10 +731,20 @@ ColumnsDescription InterpreterCreateQuery::getColumnsDescription(
     if (mode <= LoadingStrictnessLevel::SECONDARY_CREATE && !is_restore_from_backup && context_->getSettingsRef()[Setting::flatten_nested])
         res.flattenNested();
 
+    bool hasPhysicalColumns = false;
+    bool hasInsertableColumns = false;
+    for (const auto & column : res)
+    {
+        hasPhysicalColumns |= column.default_desc.kind == ColumnDefaultKind::Default || column.default_desc.kind == ColumnDefaultKind::Materialized;
+        hasInsertableColumns |= column.default_desc.kind == ColumnDefaultKind::Default || column.default_desc.kind == ColumnDefaultKind::Ephemeral;
+    }
 
-    if (res.getAllPhysical().empty())
+    if (!hasPhysicalColumns)
         throw Exception(ErrorCodes::EMPTY_LIST_OF_COLUMNS_PASSED, "Cannot CREATE table without physical columns");
 
+    if (mode <= LoadingStrictnessLevel::SECONDARY_CREATE && !is_restore_from_backup && !hasInsertableColumns)
```

Comment:
```suggestion
    if (mode <= LoadingStrictnessLevel::CREATE && !is_restore_from_backup && !hasInsertableColumns)
```

I think we still want to execute secondary create?